### PR TITLE
Use node id metedata in the attach rpc

### PR DESCRIPTION
**Pull request by rgallor**
Reflect the astarte-message-hub-proto changes: the node id information has been removed from the attach rpc parameter, which now contains only the introspection.

### CODE REVIEW

File: src/transport/grpc/mod.rs [hunk 1]
```diff
@@ -610,17 +610,17 @@ struct NodeData {
...
+        let node = Node::from_interfaces(interfaces)?;
 
         Ok(Self { node })
     }
 
-    fn try_from_interfaces(uuid: Uuid, interfaces: &Interfaces) -> Result<Self, GrpcError> {
-        Self::try_from_iter(uuid, interfaces.iter())
+    fn try_from_interfaces(interfaces: &Interfaces) -> Result<Self, GrpcError> {
+        Self::try_from_iter(interfaces.iter())
     }
```
Comment (joshuachp) [inline src/transport/grpc/mod.rs:624]: Implement TryFrom here.

File: src/transport/grpc/mod.rs [hunk 2]
```diff
@@ -649,9 +649,11 @@ mod test {
 
     use super::*;
 
+    const ID: Uuid = uuid!("67e55044-10b1-426f-9247-bb680e5fe0c8");
+
     #[derive(Debug)]
     enum ServerReceivedRequest {
-        Attach(Node),
+        Attach((Uuid, Node)),
```
Comment (harlem88) [inline src/transport/grpc/mod.rs:656]: Why did you add uuid here? Is not possible to read from `get_node_id_from_metadata`?

**Review by joshuachp — CHANGES_REQUESTED** [review]

**Review by harlem88 — APPROVED** [review]

**Review by joshuachp — APPROVED** [review]
